### PR TITLE
@joeyAghion => Mutate collector profile

### PR DESCRIPTION
**Pull request by mzikherman**
Builds upon https://github.com/artsy/metaphysics/pull/586

This lets you set the `loyalty_applicant` setting (and I threw in the `professional_buyer` as an example).

screen shot 2017-03-30 at 4 41 27 pm

### CODE REVIEW

File: schema/me/update_collector_profile.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      type: GraphQLBoolean,
+    },
+  },
+  resolve: (root, {
+    loyalty_applicant,
+    professional_buyer,
+  }, request, { rootValue: { accessToken } }) => {
+    if (!accessToken) return null;
+    return gravity.with(accessToken, {
+      method: 'POST',
```
Comment (joeyAghion) [inline schema/me/update_collector_profile.js]: Why is this update using a `POST`? I realize this particular endpoint is extra-forgiving and may update the existing `CollectorProfile` if it doesn't exist, but shouldn't this use the `PUT` endpoint instead to match the "update" intent?

**Comment by joeyAghion** [comment]
This is a new and somewhat scary [for me] new pattern in how our data is accessed and updated, but I'm curious to see how it's adopted and extended.

File: schema/me/update_collector_profile.js (hunk 1 above)
Comment (mzikherman) [inline schema/me/update_collector_profile.js]: Doh you're right! The endpoint is forgiving enough, but should totally use `PUT`.

**Comment by mzikherman** [comment]
Updated with `PUT`

**Comment by joeyAghion** [comment]
🙈

**Pull request merged**